applications/GeoMechanicsApplication/tests/cpp_tests/custom_constitutive/test_mohr_coulomb_with_tension_cutoff.cpp

```diff
@@ -283,131 +242,107 @@ KRATOS_TEST_CASE_IN_SUITE(MohrCoulombWithTensionCutOff_CalculateMaterialResponse
     KRATOS_EXPECT_VECTOR_NEAR(mapped_stress_vector, expected_cauchy_stress_vector, Defaults::absolute_tolerance);
 
     // Act
     cauchy_stress_vector <<= 14.0, 9.0, 6.0, 0.0;
-    parameters.SetStrainVector(strain_vector);
-    parameters.SetStressVector(cauchy_stress_vector);
-    law.SetValue(CAUCHY_STRESS_VECTOR, cauchy_stress_vector, process);
-    law.FinalizeMaterialResponseCauchy(parameters);
-    law.CalculateMaterialResponseCauchy(parameters);
-    law.GetValue(CAUCHY_STRESS_VECTOR, mapped_stress_vector);
+    mapped_stress_vector = CalculateMappedStressVector(cauchy_stress_vector, parameters, law);
 
     // Assert
-    expected_cauchy_stress_vector <<= 10.0, 9.0, 2.0, 0.0;
+    expected_cauchy_stress_vector <<= 10.0, 9.0, 6.0, 0.0;
     KRATOS_EXPECT_VECTOR_NEAR(mapped_stress_vector, expected_cauchy_stress_vector, Defaults::absolute_tolerance);
 
     // Act
     cauchy_stress_vector <<= 14.0, 12.0, 6.0, 0.0;
-    parameters.SetStrainVector(strain_vector);
-    parameters.SetStressVector(cauchy_stress_vector);
-    law.SetValue(CAUCHY_STRESS_VECTOR, cauchy_stress_vector, process);
-    law.FinalizeMaterialResponseCauchy(parameters);
-    law.CalculateMaterialResponseCauchy(parameters);
-    law.GetValue(CAUCHY_STRESS_VECTOR, mapped_stress_vector);
+    mapped_stress_vector = CalculateMappedStressVector(cauchy_stress_vector, parameters, law);
 
     // Assert
-    expected_cauchy_stress_vector <<= 10.0, 10.0, 0.0, 0.0;
+    expected_cauchy_stress_vector <<= 10.0, 10.0, 6.0, 0.0;
     KRATOS_EXPECT_VECTOR_NEAR(mapped_stress_vector, expected_cauchy_stress_vector, Defaults::absolute_tolerance);
 }
 
-KRATOS_TEST_CASE_IN_SUITE(MohrCoulombWithTensionCutOff_CalculateMaterialResponseCauchyWithLargeTensileStrength,
+KRATOS_TEST_CASE_IN_SUITE(MohrCoulombWithTensionCutOff_CalculateMaterialResponseCauchyAtTensionApexReturnZone,
                           KratosGeoMechanicsFastSuiteWithoutKernel)
 {
     // Arrange
     auto law = MohrCoulombWithTensionCutOff(std::make_unique<PlaneStrain>());
 
+    const auto                  properties = CreateProperties(35.0, 10.0, 20.0, 10.0);
     ConstitutiveLaw::Parameters parameters;
-    Properties                  properties;
-    properties.SetValue(GEO_FRICTION_ANGLE, 35.0);
-    properties.SetValue(GEO_COHESION, 10.0);
-    properties.SetValue(GEO_DILATANCY_ANGLE, 20.0);
-    properties.SetValue(GEO_TENSILE_STRENGTH, 20.0);
     parameters.SetMaterialProperties(properties);
-    ProcessInfo process;
+    law.InitializeMaterial(properties, {}, {});
+
+    // Act
+    Vector cauchy_stress_vector(4);
+    cauchy_stress_vector <<= 19.0, 12.0, 11.0, 0.0;
+    auto mapped_stress_vector = CalculateMappedStressVector(cauchy_stress_vector, parameters, law);
+
+    // Assert
+    Vector expected_cauchy_stress_vector(4);
+    expected_cauchy_stress_vector <<= 10.0, 10.0, 10.0, 0.0;
+    KRATOS_EXPECT_VECTOR_NEAR(mapped_stress_vector, expected_cauchy_stress_vector, Defaults::absolute_tolerance);
```

Comment:
Perhaps we can eliminate variable `mapped_stress_vector` here too, e.g.:
```suggestion
    // Act & assert
    Vector cauchy_stress_vector(4);
    cauchy_stress_vector <<= 19.0, 12.0, 11.0, 0.0;
    Vector expected_cauchy_stress_vector(4);
    expected_cauchy_stress_vector <<= 10.0, 10.0, 10.0, 0.0;
    KRATOS_EXPECT_VECTOR_NEAR(CalculateMappedStressVector(cauchy_stress_vector, parameters, law), expected_cauchy_stress_vector, Defaults::absolute_tolerance);
```

Reply:
eliminated